core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -60,6 +65,12 @@ public int formatVersion() {
     public void applyTo(TableMetadata.Builder metadataBuilder) {
       metadataBuilder.upgradeFormatVersion(formatVersion);
     }
+
+    @Override
+    public void applyTo(ImmutableViewMetadata.Builder viewMetadataBuilder) {
+      viewMetadataBuilder.formatVersion(formatVersion);
+      viewMetadataBuilder.addChanges(new MetadataUpdate.UpgradeFormatVersion(formatVersion));
```

Comment:
Changes should be handled internally to the builder, like the `TableMetadata` builder, right?

Reply:
this is a bit difficult with a generated Builder unfortunately. One approach to solve this would be to define our own `Builder` that could handle adding the right metadata changes. See the implementation details of that Builder above and let me know what you think

Reply:
this is handled in our own builder now